include/seastar/core/gate.hh

```diff
@@ -315,8 +315,7 @@ template <typename Func>
 inline
 auto
 with_gate(gate& g, Func&& func) {
```

Comment:
Maybe deprecate the with_gate() altogether? Holder provide better way of not-forgetting-to-leave the gate

Reply:
We have similar with_* functions like with_closeable, with_lock, etc. that I like for providing scoped and protected access while using some resource, and this works great for continuations.
The holder concept that is more open ended and serves RAII usage in coroutines, which aren't even enabled by default in seastar.
I think that when we enable coroutines by default it'd be a good time to consider deprecating the with_* wrappers.